common/nng/socket.cpp

```diff
@@ -39,14 +39,22 @@ Socket::~Socket()
 
 bool Socket::SyncRequired() { return false; }
 
-int Socket::Connect(const std::string &addr, bool retry)
+int Socket::Connect(const std::string &addr, int retry)
```

Comment:
I like such functions short and sweet, Linux kernel style.

How about this -
```suggestion
int Socket::Connect(const std::string &addr, unsigned max_retries)
{
    mapf_assert(!addr.empty());
    int rc, retries = 0;
    do {
        rc = nng_dial(sock->sd_, addr.c_str(), nullptr, 0);
        if (rc == 0)
            break;
        MAPF_DEBUG(rc, "nng_dial failed: " << nng_strerror(rc) << " retry #" << retries);
        msleep(10);
    } while (retry_count++  < retries);

    MAPF_ERR_IF(rc, "nng_dial failed");
    return rc;
}
```

Reply:
That looks great!

Just one further improvement: isn't the C++ way to use `std::this_thread::sleep_for` ?

Reply:
Yep, much better - I updated my proposal.

Reply:
LGTM. I'm going to press the "commit suggestion" button. Hopefully that will get committed to the branch, not to master, and @LiorAmram  can still test it...